hack/check-rec-rule-usage.sh

```diff
@@ -1,22 +1,47 @@
 #!/usr/bin/env bash
-set -e
```

Comment:
any reasons for unsetting this option? Usually when a command has an error, we always want the script to exit